frontend/packages/console-dynamic-plugin-sdk/src/api/api-types.ts

```diff
@@ -43,3 +44,18 @@ export type UseK8sWatchResources = <R extends ResourcesObject>(
 export type UseResolvedExtensions = <E extends Extension>(
   ...typeGuards: ExtensionTypeGuard<E>[]
 ) => [ResolvedExtension<E>[], boolean, any[]];
+
+/* Horizontal Nav Types */
+export type NavPage = {
+  href?: string;
+  path?: string;
+  name?: string;
+  nameKey?: string;
+  component?: React.ComponentType<RouteComponentProps>;
+};
+
+export type HorizontalNavFacadeProps = {
```

Comment:
Why export `HorizontalNavFacadeProps` as api?

Reply:
This is the props accepted as part of `HorizontalNav` perhaps I should rename it as `HorizontalNavProps` as users don't need to know about the existence of a facade. WDYT?

Reply:
We don't expose prop types. They can get the prop types using `React.ComponentProps`:
`React.ComponentProps<typeof HorizontalNav>`